src/Native/Common/pal_config.h.in

```diff
@@ -36,6 +36,9 @@
 #cmakedefine01 HAVE_TCP_VAR_H
 #cmakedefine01 HAVE_RT_MSGHDR
 #cmakedefine01 HAVE_LINUX_RTNETLINK_H
+#cmakedefine01 HAVE_INOTIFY_INIT
```

Comment:
@sokket: inotify is a Linux-specific API. My suggestion is to do the usual feature detection, then error out in CMake when configuring for Linux if the inotify APIs are not available as per the feature detection. For example, in configure.cmake:

```
check_function_exists(
    inotify_init
    HAVE_INOTIFY_INIT)

...

if (CMAKE_SYSTEM_NAME STREQUAL Linux)
    ...
    if (!HAVE_INOTIFY_INIT OR !HAVE_INOTIFY_ADD_WATCH OR !HAVE_INOTIFY_RM_WATCH)
        message(FATAL_ERROR "Cannot find inotify")
    endif()
...
endif()
```